### ISSUE TITLE
Why did you use low-pass filter twice in AMPBlock?

### ISSUE THREAD

**Issue by hoyden** [sh-lee-prml/BigVGAN#6]
https://github.com/sh-lee-prml/BigVGAN/blob/37e49f36e50134de45b407bf2c6b1a61cea09329/models_bigvgan.py#L66

Is these necessary? I didn't find it in paper.

**Comment by sh-lee-prml** [comment]
I refer to Appendix A (page 13) of BigVGAN paper.

--------------------------------------------------------------------------------------------------------------
The upsampled feature is followed by M number of AMP residual blocks,
where each AMP block uses different kernel sizes for a stack of dilated 1D convolutions defined as
ki,j (j = {1, . . . , M}).  --> M =3 [3, 7, 11]

The j-th AMP block contains L number of the anti-aliased periodic activation
and the dilated 1D convolution using a dilatation rate of di,j,l(l = {1, . . . , L}). 

--> L = 6 (3x2) [[1,1], [3,1], [5,1]]
--------------------------------------------------------------------------------------------------------------
So I used low-pass filter twice in this module.

But, I'm not sure it is same with what the author intended. When i first implemented this part, I was also confused because the figure was not matched with the hyperparameter (Specifically, dilation rates).

**Comment by hoyden** [comment]
Thanks for your reply.

**Comment by sh-lee-prml** [comment]
Thank you for your concern about this issue.

I agree your idea about using a low-pass filter only once. However, in this case, I was confused about activation function between these dilated convs.

The original HiFi-GAN used a leaky-relu for activation function in this part. However, in this paper, BigVGAN replaced it with snake1d. Hence, I implemented this part just by using snake and resamping with low-pass filter twice.

I just think that it would have been nice if there are additional ablation studies about this part in this paper.

I'm not sure that which implementation is best in this cases. I hope that the authors of BigVGAN adress thier idea with more ablation studies...

**Comment by hoyden** [comment]
Yeah, you're right. I'll try to use snake1d directly between two dilated convs. And I will tell you the results of my experiment. Maybe someday NVIDIA will open source their work. ^^

**Comment by sh-lee-prml** [comment]
After some comparison, I found that both models (using resampling once or twice) have similar performance. 

But when using twice, training/inference speed is much slower so I changed it as you mentioned

**Comment by rarents1975** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
lsiqueira, @bitsark, @joytianya, @ghchris2021, @RUI-LONG, @wangjing0, @jmetzz, @davoclavo, @tchaton, @Nyandwi, @pikid, @AlexeyAB, @askmetoo, @malteos, @DeShuiYu, @eugenesiow, @Santinule, @kennethleungty, @sumandas0, @NaoMatch
